frontend/src/metabase/query_builder/actions/visualization-settings.ts

```diff
@@ -24,16 +24,11 @@ export const onUpdateVisualizationSettings =
       queryBuilderMode === "dataset" && datasetEditorTab === "metadata";
     const wasJustEditingModel =
       previousQueryBuilderMode === "dataset" && queryBuilderMode !== "dataset";
-    const changedSettings = Object.keys(settings);
-    const isColumnWidthResetEvent =
-      changedSettings.length === 1 &&
-      changedSettings.includes("table.column_widths") &&
-      settings["table.column_widths"] === undefined;
```

Comment:
We extract these conditions into `shouldUpdateVisualizationSettings`, so it can be re-used outside of this Redux action.